cobalt/browser/metrics/cobalt_metrics_services_manager_client_test.cc

```diff
@@ -0,0 +1,256 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/browser/metrics/cobalt_metrics_services_manager_client.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/command_line.h"
+#include "base/files/scoped_temp_dir.h"
+#include "base/path_service.h"
+#include "base/test/scoped_path_override.h"
+#include "base/test/task_environment.h"
+#include "cobalt/browser/metrics/cobalt_enabled_state_provider.h"
+#include "cobalt/browser/metrics/cobalt_metrics_service_client.h"
+#include "components/metrics/metrics_pref_names.h"
+#include "components/metrics/metrics_service.h"
+#include "components/metrics/metrics_state_manager.h"
+#include "components/metrics/metrics_switches.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/testing_pref_service.h"
+#include "components/variations/service/variations_service.h"
+#include "components/variations/synthetic_trial_registry.h"
+#include "content/shell/browser/shell_paths.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace cobalt {
+
+class CobaltMetricsServicesManagerClientTest : public ::testing::Test {
+ protected:
+  CobaltMetricsServicesManagerClientTest() = default;
+
+  void SetUp() override {
+    temp_dir_.CreateUniqueTempDir();
+
+    path_override_ = std::make_unique<base::ScopedPathOverride>(
+        content::SHELL_DIR_USER_DATA, temp_dir_.GetPath());
+
+    metrics::MetricsService::RegisterPrefs(prefs_.registry());
+    // Add any other prefs that might be accessed.
+    prefs_.registry()->RegisterBooleanPref(
+        metrics::prefs::kMetricsReportingEnabled, false);
+
+    // Make sure user_metrics.cc has a g_task_runner.
+    base::SetRecordActionTaskRunner(
+        task_environment_.GetMainThreadTaskRunner());
+
+    manager_client_ =
+        std::make_unique<CobaltMetricsServicesManagerClient>(&prefs_);
+  }
+
+  void TearDown() override {
+    manager_client_.reset();
+    path_override_.reset();
+    base::CommandLine::ForCurrentProcess()->RemoveSwitch(
+        metrics::switches::kForceEnableMetricsReporting);
```

Comment:
nit: Is this needed? I see flag being manipulated only
in `MetricsReportingIsEnabledByForceSwitch`.

Reply:
Not particularly. Removed.